Reconstruction/doc/RecCalorimeter.md

```diff
@@ -3,51 +3,102 @@ RecCalorimeter package
-* fieldNames (vector of string, name of the fields describing the segmented volume)
-* fieldValues (vector of int, values of the fields describing the segmented volume);
+### 1. Create calorimeter towers.
 
-More details can be found in [Doxygen documentation] (http://fccsw.web.cern.ch/fccsw/).
+A tower contains all cells within certain eta and phi (tower size: **deltaEtaTower**, **deltaPhiTower**). Distance in r plays no role in this algorithm.
+
+>TODO: Currently there is no support of cell splitting, so each cell should be completely inside the tower and that can be achieved using `GridEtaPhi` segmentation.
+
+### 2. Find local maxima.
+
+Local maxima are found using the sliding window of a fixed size in eta x phi (**nEtaWindow** **nPhiWindow** in units of tower size). If a local max is found and its energy is above threshold (**energyThreshold**), it is added to the preclusters list. Each precluster contains the barycentre position and the transverse energy. Position is recalculated using the window size in eta x phi (**nEtaPosition**, **nPhiPosition**) that may be smaller than the sliding window to reduce the noise influence. Both windows are centred at the same tower. The energy of the precluster is the energy calculated using the sliding window.
+
+### 3. Remove duplicates.
+
+If two pre-clusters are found next to each other (within window **nEtaDuplicates**, **nPhiDuplicates**), the pre-cluster with lower energy is removed.
+
+>TODO: Currently there is no support on energy sharing between clusters, so if the duplicate window is smaller than sliding window, some cells may be taken twice (instead of the weighting their energies).
+
+### 4. Build clusters.
+
+Clusters are created using the pre-clusters energy (energy of towers within the sliding window). Position is calculated from the barycentre position and the inner radius of the detector. For each cluster the cell collection is searched and all those inside the cluster are attached.
+
+>TODO: Final cluster may have a different size, so recalculation of energy around the tower containing barycentre should be done.
 
 # Example
 
-Example script which runs ECAL cell reconstruction is [here] (https://github.com/HEP-FCC/FCCSW/blame/master/Reconstruction/RecCalorimeter/options/runEcalReconstruction.py).
+Example script which runs ECAL reconstruction is [here](../RecCalorimeter/options/runEcalReconstruction.py).
 
 * Read input file with Geant4 hits
 * Produce calo cells (CreateCaloCells)
 * Store cells as CaloHits with cellID (produced by CreateCaloCells) and as CaloClusters with position (produced by CreatePositionedHit)
+* Reconstruct clusters using the sliding window algorithm (CreateCaloClustersSlidingWindow)
+
+First, prepare the input file. Before running the script, load the library libDetSegmentation.so (necessary because of the phi-eta segmenation usage):
+~~~{.sh}
+LD_PRELOAD=build.$BINARY_TAG/lib/libDetSegmentation.so
```

Comment:
It is necessary if we use our own segmentation (not part of DD4hep). We will get rid of that eventually - segmentation will move to DD4hep and/or DD4hep will handle segmentation using root plugin as other things: [JIRA ticket](https://sft.its.cern.ch/jira/browse/DDFORHEP-24).